kotlinx-collections-immutable/tests/src/test/kotlin/kotlinx.collections.immutable/contractTests/GuavaImmutableCollectionBaseTest.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kotlinx.collections.immutable.contractTests
+
+import junit.framework.AssertionFailedError
+import junit.framework.TestListener
+import junit.framework.TestResult
+import junit.framework.TestSuite
+import java.io.PrintWriter
+import java.io.StringWriter
+import kotlin.test.fail
+
+open class GuavaImmutableCollectionBaseTest: TestListener {
+    override fun addFailure(test: junit.framework.Test, e: AssertionFailedError) {
```

Comment:
Why not just rethrow these exceptions here?

Reply:
+1